packages/fast-components-react-msft/src/action-toggle/action-toggle.tsx

```diff
@@ -0,0 +1,220 @@
+        } else {
+            return this.renderUnselectedGlyph();
+        }
+    }
+
+    public renderSelectedGlyph(): React.ReactNode {
+        if (typeof this.props.selectedGlyph === "function") {
+            return this.props.selectedGlyph(
+                get(this.props, "managedClasses.actionToggle_selectedGlyph")
+            );
+        }
+        return null;
+    }
+
+    public renderUnselectedGlyph(): React.ReactNode {
+        if (typeof this.props.unselectedGlyph === "function") {
+            return this.props.unselectedGlyph(
+                get(this.props, "managedClasses.actionToggle_unselectedGlyph")
+            );
+        }
+        return null;
+    }
+
+    /**
+     * Generates class names
+     */
+    protected generateClassNames(): string {
+        let classNames: string = get(this.props, "managedClasses.actionToggle") || "";
+
+        if (this.props.disabled) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__disabled"
+            )}`;
+        }
+
+        if (this.state.selected) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__selected"
+            )}`;
+            classNames = `${classNames} ${this.getAppearanceString(
+                ActionToggleAppearance.primary
+            )}`;
+        } else {
+            classNames = `${classNames} ${this.getAppearanceString(
+                ActionToggleAppearance.lightweight
+            )}`;
+        }
+
+        if (this.hasGlyphAndContent()) {
+            classNames = `${classNames} ${get(
+                this.props,
+                "managedClasses.actionToggle__hasGlyphAndContent"
+            )}`;
+        }
+
+        return super.generateClassNames(classNames);
+    }
+
+    private getAppearanceString(appearance: ActionToggleAppearance): string {
+        switch (appearance) {
+            case ActionToggleAppearance.primary:
+                return get(this.props, "managedClasses.actionToggle__primary");
+            case ActionToggleAppearance.lightweight:
+                return get(this.props, "managedClasses.actionToggle__lightweight");
+        }
+    }
+
+    /**
+     * Checks to see if the toggle is displaying both glyph and content or not
+     */
+    private hasGlyphAndContent(): boolean {
+        return this.state.selected
+            ? !isNullOrUndefined(this.props.selectedGlyph) &&
+                  !isNullOrUndefined(this.props.selectedContent)
+            : !isNullOrUndefined(this.props.unselectedGlyph) &&
+                  !isNullOrUndefined(this.props.unselectedContent);
+    }
+
+    /**
+     * Handles onClick
+     */
+    private handleToggleChange = (e: React.MouseEvent<HTMLElement>): void => {
+        if (typeof this.props.selected !== "boolean") {
+            this.setState({ selected: !this.state.selected });
+        }
+        if (this.props.onChange) {
```

Comment:
I think you'll need to check if this is a function

Reply:
yep